### DISCUSSION TITLE
session idle timeout for Trino

### DISCUSSION THREAD

**Discussion by wetcod** [trinodb/trino#23828] [Q&A]
Hello,

Is there any setting related to session idle timeout in trino?

If the session is maintained without disconnecting during data fetch, how long will it be maintained?

I'm looking for a setting that is equivalent to `hive.server2.idle.session.timeout` of Hive.

**Comment by Heltman** [comment]
query.client.timeout[#](https://trino.io/docs/current/admin/properties-query-management.html#query-client-timeout)
Type: [duration](https://trino.io/docs/current/admin/properties.html#prop-type-duration)

Default value: 5m

Configures how long the cluster runs without contact from the client application, such as the CLI, before it abandons and cancels its work.

**Comment by hashhar** [comment]
In addition to what is said above I'll say the Trino protocol doesn't need persistent connections so there's no "session" to speak of in that sense.

When a query is not executing there's no active connection to the cluster. Similarly if you are using a client and not actively fetching more rows there's no ACTIVE connection (other than for connection reuse purposes for example).

**Comment by pavels1983** [comment]
As trino does not need persistent connections (no session), how does authentication works, it repeats for each query?

**Comment by hashhar** [comment]
> As trino does not need persistent connections (no session), how does authentication works, it repeats for each query?

The HTTP connection is usually re-used so not need for repeated authentications (in most client implementations).
There being no session doesn't mean there isn't a persistent HTTP connection.

If you use `curl` locally for example (or change some client to not reuse connections) then yes you'll notice authentication being repeated for every new connection being opened.
